OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -14225,6 +14239,12 @@ algorithm
         local Integer index;
         case SOME(index)
         then SOME(index + getScalarElementIndex(subs, List.map(sv.numArrayElement, stringInt)) - 1);
+        // PHI: NONE() needs to be caught or else indexed arrayVars for the jacobian won't be found
+        // (things like $DER.a[$i].x.$pDERA.dummyVarA with 'x' being a state inside array 'a')
+        // not sure what the actual problem is, i.e. why they have no variable_index
+        case NONE() guard not Flags.isSet(Flags.NF_SCALARIZE) //algorithm
+          //print("simVarFromHT: sv has no variable_index: " + ComponentReference.printComponentRefStr(inCref) + "\n");
```

Comment:
Leftover debug code? Add  protected variable `Boolean debug = false` and a `if debug then ...` around the print if you want to keep it.

Reply:
Could this be related to the index problem in `testArrayEquations.mos`? I'll have a closer look at all those indices.